src/main/java/com/alvarium/contracts/Annotation.java

```diff
@@ -0,0 +1,129 @@
+package com.alvarium.contracts;
+
+
+import java.io.Serializable;
+import java.time.Instant;
+
+import com.alvarium.hash.HashType;
+import com.alvarium.serializers.InstantConverter;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import de.huxhorn.sulky.ulid.ULID;
+
+/**
+ * A java bean that encapsulates all of the data related to a specific annotation.
+ * this will be generated by the annotators.
+ */
+public class Annotation implements Serializable {
```

Comment:
Yea that generally makes sense. I was unsure at first which fields are required to be set explicitly by the annotators, so I've created the setters just in case. I'll take a look into the domain model that you've mentioned when we start working on the annotators. Another simple solution, in my opinion, is to apply the builder pattern on the annotation class to preserve immutability within the annotation. 

For now, I've removed all the setters from the the Annotation class, and changed all fields to be `final` and are set explicitly through the constructor to preserve immutability. We'll see how we'll proceed once this PR is merged.

Reply:
fixed in 68a8871.